src/RefractAST.cc

```diff
@@ -719,15 +670,30 @@ namespace drafter
                 throw std::runtime_error("Unhandled type of DataStructure");
         }
 
+        return element;
+    }
+
+    sos::Object SerializeRefract(refract::IElement* element, const refract::Registry& registry) {
+
         if (!element) {
             return sos::Object();
         }
 
+        refract::ExpandVisitor expander(registry);
```

Comment:
I am feeling really bad about doing the expansion while serializing. It is going to introduce a lot of duplicate code. Can we separate the expansion part out of the serialization?